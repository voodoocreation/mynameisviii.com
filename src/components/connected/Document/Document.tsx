import { IncomingMessage } from "http";
import Document, {
  DocumentContext,
  DocumentProps,
  Head,
  Main,
  NextScript
} from "next/document";
import * as React from "react";

const isDev = () => process.env.NODE_ENV !== "production";

const Meta: React.FC = () => (
  <React.Fragment>
    <meta charSet="UTF-8" />
    <meta
      content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
      name="viewport"
    />

    <meta property="fb:app_id" content="278225392903510" />
    <meta property="fb:page_id" content="218519344943411" />

    <link rel="manifest" href="/static/manifest.json" />

    <link
      rel="apple-touch-icon"
      sizes="180x180"
      href="/static/favicon/apple-touch-icon.png"
    />
    <link
      rel="icon"
      type="image/png"
      sizes="32x32"
      href="/static/favicon/favicon-32x32.png"
    />
    <link
      rel="icon"
      type="image/png"
      sizes="16x16"
      href="/static/favicon/favicon-16x16.png"
    />
    <link
      rel="mask-icon"
      href="/static/favicon/safari-pinned-tab.svg"
      color="#000000"
    />
    <link rel="shortcut icon" href="/static/favicon/favicon.ico" />
    <meta name="msapplication-TileColor" content="#000000" />
    <meta name="msapplication-config" content="/static/browserconfig.xml" />
    <meta name="theme-color" content="#ffffff" />
  </React.Fragment>
);

const AnalyticsHead: React.FC = () => (
  <script
    dangerouslySetInnerHTML={{
      __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
        'https://www.googletagmanager.com/gtm.js?id='+i+dl${
          isDev()
            ? `+ '&gtm_auth=5N0VfAOq-iEoyMOidFx0kQ&gtm_preview=env-3&gtm_cookies_win=x'`
            : ""
        };f.parentNode.insertBefore(j,f);
      })(window,document,'script','dataLayer','GTM-TJZ52XC');`
    }}
  />
);

const AnalyticsBody: React.FC = () => (
  <noscript>
    <iframe
      src={`https://www.googletagmanager.com/ns.html?id=GTM-TJZ52XC${
        isDev()
          ? `&gtm_auth=5N0VfAOq-iEoyMOidFx0kQ&gtm_preview=env-3&gtm_cookies_win=x`
          : ""
      }`}
      height="0"
      width="0"
      style={{ display: "none", visibility: "hidden" }}
    />
  </noscript>
);

interface IProps extends DocumentProps {
  locale: string;
}

export default class<P extends IProps> extends Document<P> {
  public static async getInitialProps(context: DocumentContext) {
    const initialProps = await Document.getInitialProps(context);
    const props = await context.renderPage();
    const req = context.req as IncomingMessage & {
      locale: string;
    };

    return {
      ...initialProps,
      ...props,
      locale: req.locale || "en-NZ"
    };
  }

  public render() {
    return (
      <html lang={this.props.locale}>
        <Head>
          <AnalyticsHead />
          <script
            dangerouslySetInnerHTML={{
              __html: `document.documentElement.classList.add("isClientRendered");`
            }}
          />
          <script src="https://maps.googleapis.com/maps/api/js?v=3.exp&key=AIzaSyBO41slvNUu9QtGI9aNZ4RLeqBlD13y_5M" />

          <Meta />
        </Head>
        <body>
          <AnalyticsBody />
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
