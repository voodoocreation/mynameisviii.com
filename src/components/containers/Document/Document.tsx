import { IncomingMessage } from "http";
import Document, {
  Head,
  Main,
  NextDocumentContext as Context,
  NextScript
} from "next/document";
import * as React from "react";

const isDev = () => process.env.NODE_ENV !== "production";

// tslint:disable-next-line
const css = require("../../../scss/index.scss");

const Meta: React.SFC<{}> = () => (
  <React.Fragment>
    <meta charSet="UTF-8" />
    <meta content="width=device-width, initial-scale=1.0" name="viewport" />

    <meta name="theme-color" content="#160f20" />
    <meta name="msapplication-TileColor" content="#160f20" />
    <meta
      name="msapplication-TileImage"
      content="/static/favicon/mstile-150x150.png"
    />
    <link rel="icon" href="/static/favicon/favicon.ico" />
    <link
      rel="apple-touch-icon"
      sizes="180x180"
      href="/static/favicon/apple-touch-icon.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="152x152"
      href="/static/favicon/apple-touch-icon-152x152.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="144x144"
      href="/static/favicon/apple-touch-icon-144x144.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="120x120"
      href="/static/favicon/apple-touch-icon-120x120.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="114x114"
      href="/static/favicon/apple-touch-icon-114x114.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="76x76"
      href="/static/favicon/apple-touch-icon-76x76.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="72x72"
      href="/static/favicon/apple-touch-icon-72x72.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="60x60"
      href="/static/favicon/apple-touch-icon-60x60.png"
    />
    <link
      rel="apple-touch-icon"
      sizes="57x57"
      href="/static/favicon/apple-touch-icon-57x57.png"
    />
    <link
      rel="icon"
      type="image/png"
      sizes="32x32"
      href="/static/favicon/favicon-32x32.png"
    />
    <link
      rel="icon"
      type="image/png"
      sizes="16x16"
      href="/static/favicon/favicon-16x16.png"
    />
    <link
      rel="mask-icon"
      href="/static/favicon/safari-pinned-tab.svg"
      color="#160f20"
    />
    <link rel="manifest" href="/static/manifest.json" />
  </React.Fragment>
);

const AnalyticsHead: React.SFC<{}> = () => (
  <script
    dangerouslySetInnerHTML={{
      __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
        'https://www.googletagmanager.com/gtm.js?id='+i+dl${
          isDev()
            ? `+ '&gtm_auth=5N0VfAOq-iEoyMOidFx0kQ&gtm_preview=env-3&gtm_cookies_win=x'`
            : ""
        };f.parentNode.insertBefore(j,f);
      })(window,document,'script','dataLayer','GTM-TJZ52XC');`
    }}
  />
);

const AnalyticsBody: React.SFC<{}> = () => (
  <noscript>
    <iframe
      src={`https://www.googletagmanager.com/ns.html?id=GTM-TJZ52XC${
        isDev()
          ? `&gtm_auth=5N0VfAOq-iEoyMOidFx0kQ&gtm_preview=env-3&gtm_cookies_win=x`
          : ""
      }`}
      height="0"
      width="0"
      style={{ display: "none", visibility: "hidden" }}
    />
  </noscript>
);

export default class extends Document {
  public static async getInitialProps(context: Context) {
    const props = await context.renderPage();
    const req = context.req as IncomingMessage & {
      intlMessages: {};
      locale: string;
    };

    return {
      ...props,
      intlMessages: req.intlMessages,
      locale: req.locale
    };
  }

  public render() {
    const styles =
      process.env.NODE_ENV !== "production" ? (
        <style dangerouslySetInnerHTML={{ __html: css }} />
      ) : (
        <link
          href={`/assets/main.css?${this.props.__NEXT_DATA__.buildId}`}
          rel="stylesheet"
        />
      );

    return (
      <html lang={this.props.locale}>
        <Head>
          <script
            dangerouslySetInnerHTML={{
              __html: `document.documentElement.classList.add("isClientRendered");`
            }}
          />
          <AnalyticsHead />
          <script src="https://maps.googleapis.com/maps/api/js?v=3.exp&key=AIzaSyDPZBC30DN_SjkhZRMd1AX8MscrqdVsQbE" />

          <Meta />

          {styles}
        </Head>
        <body>
          <AnalyticsBody />
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
