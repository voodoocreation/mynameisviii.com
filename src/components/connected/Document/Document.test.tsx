import { render } from "enzyme";
import merge from "lodash.merge";
import * as React from "react";

import Document from "./Document";

const setup = async (fn: any, fromTestProps?: any) => {
  const context = merge(
    {
      __NEXT_DATA__: {
        buildId: "buildId",
        page: "/",
        pathname: "pathname"
      },
      assetPrefix: "/assetPrefix",
      dynamicImports: [],
      files: [],
      isServer: true,
      renderPage: async () => ({}),
      req: {
        locale: "en-NZ"
      }
    },
    fromTestProps
  );
  const initialProps = await Document.getInitialProps(context);
  const props = {
    ...context,
    ...initialProps
  };

  return {
    props,
    wrapper: fn(<Document {...props} />)
  };
};

const g: any = global;

describe("[connected] <Document />", () => {
  beforeEach(() => {
    // @ts-ignore-next-line
    process.env.NODE_ENV = "development";
    g.isServer = true;
  });

  it("renders correctly when NODE_ENV=development", async () => {
    // @ts-ignore-next-line
    process.env.NODE_ENV = "development";
    const { wrapper } = await setup(render);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders currectly when NODE_ENV=production", async () => {
    // @ts-ignore-next-line
    process.env.NODE_ENV = "production";
    const { wrapper } = await setup(render);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders currectly when locale is missing", async () => {
    const { wrapper } = await setup(render, {
      req: { locale: "" }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
