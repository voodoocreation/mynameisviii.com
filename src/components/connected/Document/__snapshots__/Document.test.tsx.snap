// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[containers] <Document /> renders correctly when NODE_ENV=development 1`] = `
Array [
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/index.js"
    rel="preload"
  />,
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/_app.js"
    rel="preload"
  />,
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/_error.js"
    rel="preload"
  />,
  null,
  null,
  null,
  <meta
    charset="UTF-8"
  />,
  <meta
    content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
    name="viewport"
  />,
  <meta
    content="278225392903510"
    property="fb:app_id"
  />,
  <meta
    content="218519344943411"
    property="fb:page_id"
  />,
  <meta
    content="#160f20"
    name="theme-color"
  />,
  <meta
    content="#160f20"
    name="msapplication-TileColor"
  />,
  <meta
    content="/static/favicon/mstile-150x150.png"
    name="msapplication-TileImage"
  />,
  <link
    href="/static/favicon/favicon.ico"
    rel="icon"
  />,
  <link
    href="/static/favicon/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/static/favicon/apple-touch-icon-152x152.png"
    rel="apple-touch-icon"
    sizes="152x152"
  />,
  <link
    href="/static/favicon/apple-touch-icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/static/favicon/apple-touch-icon-120x120.png"
    rel="apple-touch-icon"
    sizes="120x120"
  />,
  <link
    href="/static/favicon/apple-touch-icon-114x114.png"
    rel="apple-touch-icon"
    sizes="114x114"
  />,
  <link
    href="/static/favicon/apple-touch-icon-76x76.png"
    rel="apple-touch-icon"
    sizes="76x76"
  />,
  <link
    href="/static/favicon/apple-touch-icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/static/favicon/apple-touch-icon-60x60.png"
    rel="apple-touch-icon"
    sizes="60x60"
  />,
  <link
    href="/static/favicon/apple-touch-icon-57x57.png"
    rel="apple-touch-icon"
    sizes="57x57"
  />,
  <link
    href="/static/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/static/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    color="#160f20"
    href="/static/favicon/safari-pinned-tab.svg"
    rel="mask-icon"
  />,
  <link
    href="/static/manifest.json"
    rel="manifest"
  />,
  null,
  <noscript>
    &lt;iframe src="https://www.googletagmanager.com/ns.html?id=GTM-TJZ52XC&amp;gtm_auth=5N0VfAOq-iEoyMOidFx0kQ&amp;gtm_preview=env-3&amp;gtm_cookies_win=x" height="0" width="0" style="display:none;visibility:hidden"&gt;&lt;/iframe&gt;
  </noscript>,
  <div
    id="__next"
  />,
  null,
  null,
  null,
  null,
]
`;

exports[`[containers] <Document /> renders currectly when NODE_ENV=production 1`] = `
Array [
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/index.js"
    rel="preload"
  />,
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/_app.js"
    rel="preload"
  />,
  <link
    as="script"
    href="/assetPrefix/_next/static/buildId/pages/_error.js"
    rel="preload"
  />,
  null,
  null,
  null,
  <meta
    charset="UTF-8"
  />,
  <meta
    content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
    name="viewport"
  />,
  <meta
    content="278225392903510"
    property="fb:app_id"
  />,
  <meta
    content="218519344943411"
    property="fb:page_id"
  />,
  <meta
    content="#160f20"
    name="theme-color"
  />,
  <meta
    content="#160f20"
    name="msapplication-TileColor"
  />,
  <meta
    content="/static/favicon/mstile-150x150.png"
    name="msapplication-TileImage"
  />,
  <link
    href="/static/favicon/favicon.ico"
    rel="icon"
  />,
  <link
    href="/static/favicon/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/static/favicon/apple-touch-icon-152x152.png"
    rel="apple-touch-icon"
    sizes="152x152"
  />,
  <link
    href="/static/favicon/apple-touch-icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/static/favicon/apple-touch-icon-120x120.png"
    rel="apple-touch-icon"
    sizes="120x120"
  />,
  <link
    href="/static/favicon/apple-touch-icon-114x114.png"
    rel="apple-touch-icon"
    sizes="114x114"
  />,
  <link
    href="/static/favicon/apple-touch-icon-76x76.png"
    rel="apple-touch-icon"
    sizes="76x76"
  />,
  <link
    href="/static/favicon/apple-touch-icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/static/favicon/apple-touch-icon-60x60.png"
    rel="apple-touch-icon"
    sizes="60x60"
  />,
  <link
    href="/static/favicon/apple-touch-icon-57x57.png"
    rel="apple-touch-icon"
    sizes="57x57"
  />,
  <link
    href="/static/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/static/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    color="#160f20"
    href="/static/favicon/safari-pinned-tab.svg"
    rel="mask-icon"
  />,
  <link
    href="/static/manifest.json"
    rel="manifest"
  />,
  <link
    href="/assets/main.css"
    rel="stylesheet"
  />,
  <noscript>
    &lt;iframe src="https://www.googletagmanager.com/ns.html?id=GTM-TJZ52XC" height="0" width="0" style="display:none;visibility:hidden"&gt;&lt;/iframe&gt;
  </noscript>,
  <div
    id="__next"
  />,
  null,
  null,
  null,
  null,
]
`;
